tululbot/utils.py

```diff
@@ -20,9 +40,35 @@ def retrieve_random(self):
     def format_quote(self, q):
         return '{q[quote]} - {q[author]}, {q[author_bio]}'.format(q=q)
 
-    def refresh_cache(self):
+    def refresh_cache_if_applicable(self):
+        if (not self.refresh_url_if_applicable()):
+            return False
+
         body = requests.get(self._quote_url).text
-        # What if previosuly we have the cache, but this time
+        # What if previously we have the cache, but this time
         # when we try to get new cache, the network occurs error?
         # We will think about "don't refresh if error" later.
         self._cache = yaml.load(body)['quotes']
+
+        return True
+
+    def refresh_url_if_applicable(self):
+        now = TimeHelper.now()
+        delta = now - self._last_refresh
+
+        if (delta.seconds < self._refresh_interval):
+            return False
+
+        res = requests.get(self._git_branch_check_url)
+
+        # Don't care broken request
+        if (not res.ok):
+            return False
+
+        json = res.get_json()
+        sha = json['commit']['sha']
+        self._quote_url = 'https://cdn.rawgit.com/tulul/tulul-quotes/{}/quote.yaml'.format(sha)
+
+        self._last_refresh = now
+
+        return True
```

Comment:
Hmm now this function has side-effect, but its return value is also needed. quite problematic. how about changing the flow to something like this:

```python
def refresh_cache_if_applicable(self):
    if self.need_refresh():
        self.refresh_url()
        self.refresh_cache()

def need_refresh(self):
    return delta.seconds < self._refresh_interval

def refresh_url(self):
    # do refresh url here and don't return

def refresh_cache(self):
    # do refresh cache here and don't return
```

this way it's clear that `need_refresh` is pure while the others aren't.